src/App.tsx

```diff
@@ -1,61 +1,146 @@
 import './App.scss';
+import {
+  Button,
+  Paper,
+  FormControl,
+  TextField,
+  MenuItem,
+} from '@mui/material';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+import React, { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoWithUser } from './types/TodoWithUser';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+const getUserById = (id: number) => {
+  return usersFromServer.find(user => user.id === id) || null;
+};
+
+const todosWithUsers: TodoWithUser[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(0);
+  const [todos, setTodos] = useState(todosWithUsers);
+  const [isTitleEmpty, setIsTitleEmpty] = useState(false);
+  const [isUserNotSelected, setIsUserNotSelected] = useState(false);
+
+  const changeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setIsTitleEmpty(false);
```

Comment:
Why do we need to set this variable to false all the time? Maybe, let's do it when `isTitleEmpty` is true ?